src/vs/workbench/parts/files/browser/fileActions.ts

```diff
@@ -1783,6 +1783,79 @@ export class OpenResourcesAction extends Action {
 	}
 }
 
+export class ReopenClosedFileAction extends Action {
+
+	public static ID = 'workbench.files.action.reopenClosedFile';
+	public static LABEL = nls.localize('reopenClosedFile', "Reopen Closed File");
+
+	constructor(
+		id: string,
+		label: string,
+		@IPartService private partService: IPartService,
+		@IWorkbenchEditorService private editorService: IWorkbenchEditorService,
+		@IViewletService private viewletService: IViewletService,
+		@ITextFileService private textFileService: ITextFileService,
+		@IFileService private fileService: IFileService
+	) {
+		super(id, label);
+	}
+
+	public run(): TPromise<any> {
+		let viewletPromise = TPromise.as(null);
+		if (!this.partService.isSideBarHidden()) {
+			viewletPromise = this.viewletService.openViewlet(Files.VIEWLET_ID, false);
+		}
+
+		return viewletPromise.then(() => {
+			let workingFilesModel: Files.IWorkingFilesModel = this.textFileService.getWorkingFilesModel();
+			let resources: Files.IWorkingFileEntry[] = workingFilesModel.popLastClosedEntry();
+
+			if (resources === null) {
+				return TPromise.as(true);
+			}
+
+			let resolveFilePromises: TPromise<any>[] = resources.map((r) => { return this.fileService.resolveFile(r.resource); });
+
+			return TPromise.join(resolveFilePromises).then(() => {
+				return this.openEntries(resources, workingFilesModel);
+			}, (e: any[]) => {
+
+				// Don't attempt to reopen files that fail because they no longer exist
+				let exceptions = [];
+				e.forEach(err => {
+					if (err.code === 'ENOENT') {
```

Comment:
Oh right, this is something I wanted to add actually, because it is just an oversight. Maybe you could wire it up in this PR? Otherwise leave your current solution in and I can change it later.